Frameworks/Foundation/NSProxy.mm

```diff
@@ -14,220 +14,317 @@
+*/
+- (NSString*)debugDescription {
+    return [self description];
 }
 
+#pragma endregion
+
+#pragma region NSObject protocol functions
+
+// clang format doesn't play well with 'class' here, so turn it off
+// clang-format off
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
-+ (Class) class {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (Class)class {
+    return object_getClass(self);
 }
 
-    /**
-     @Status Stub
-     @Notes
-    */
-    - (Class) class {
-    UNIMPLEMENTED();
-    return StubReturn();
+/**
+ @Status Interoperable
+ @Notes
+*/
+- (Class)superclass {
+    return class_getSuperclass(object_getClass(self));
 }
 
-    /**
-     @Status Stub
-     @Notes
-    */
-    - (BOOL)isEqual : (id)anObject {
-    UNIMPLEMENTED();
-    return StubReturn();
+// clang-format on
+
+/**
+ @Status Interoperable
+ @Notes
+*/
+- (BOOL)isEqual:(id)anObject {
+    return self == anObject;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
+*/
+- (NSUInteger)hash {
+    return reinterpret_cast<NSUInteger>(self);
+}
+
+/**
+ @Status Interoperable
  @Notes
 */
 - (instancetype)self {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return (id)self;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (BOOL)isKindOfClass:(Class)aClass {
-    UNIMPLEMENTED();
-    return StubReturn();
+    Class forwardClass = static_cast<Class>([self performSelector:@selector(class)]);
+    return [forwardClass isSubclassOfClass:aClass];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (BOOL)isMemberOfClass:(Class)aClass {
-    UNIMPLEMENTED();
-    return StubReturn();
+    Class forwardClass = static_cast<Class>([self performSelector:@selector(class)]);
+    return aClass == forwardClass;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (BOOL)respondsToSelector:(SEL)aSelector {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self methodSignatureForSelector:aSelector] != nil;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (BOOL)conformsToProtocol:(Protocol*)aProtocol {
-    UNIMPLEMENTED();
-    return StubReturn();
+    Class forwardClass = static_cast<Class>([self performSelector:@selector(class)]);
+    return class_conformsToProtocol(forwardClass, aProtocol);
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (id)performSelector:(SEL)aSelector {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSInvocation* invocation = [NSInvocation invocationWithMethodSignature:[self methodSignatureForSelector:aSelector]];
+    invocation.selector = aSelector;
+    [self forwardInvocation:invocation];
+
+    id ret;
+    [invocation getReturnValue:&ret];
+    return ret;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (id)performSelector:(SEL)aSelector withObject:(id)anObject {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSInvocation* invocation = [NSInvocation invocationWithMethodSignature:[self methodSignatureForSelector:aSelector]];
```

Comment:
It seems like `performSelector:` and friends _could_ just call objc_msgSend; that would go through the invocation forwarding and capture codepath

It's less verbose, but perhaps there is an advantage to creating the invocation directly!